pkg/operator/highcpuusagealertcontroller/performanceprofile.go

```diff
@@ -0,0 +1,69 @@
+package highcpuusagealertcontroller
+
+import (
+	"context"
+	"encoding/json"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/utils/cpuset"
+)
+
+// default and taken from the docs
+const defaultCoresNum = 8
+
+var performanceGroup = schema.GroupVersionResource{Group: "performance.openshift.io", Version: "v2", Resource: "performanceprofiles"}
+
+type performanceProfileList struct {
+	Items []performanceProfile `json:"items"`
+}
+
+type performanceProfile struct {
+	Spec struct {
+		CPU *struct {
+			Reserved *string `json:"reserved"`
+		} `json:"cpu"`
+		NodeSelector map[string]string `json:"nodeSelector"`
+	} `json:"spec"`
+}
+
+func performanceProfileControlPlaneCores(ctx context.Context, cl dynamic.Interface) (int, error) {
```

Comment:
you could change the signature of this method to return a boolean to know if the value was found or not so we could apply the default value.